### PR TITLE
Use --dev in the Makefile

### PR DESCRIPTION
**Pull request by samoht**
This turns-on warning-as-error in developement mode.

### CODE REVIEW

File: proj/src/sub2/bin/foo_main.ml [hunk 1]
```diff
@@ -25,7 +25,7 @@ let subcommands = [
 
 let help () =
   let subcommand_names =
-    String.concat "\n" (List.map (fun (name, f) -> "  " ^ name) subcommands)
+    String.concat "\n" (List.map (fun (name, _f) -> "  " ^ name) subcommands)
```
Comment (mjambon) [inline proj/src/sub2/bin/foo_main.ml]: Hmm, I don't like this style very much.

Comment (mjambon) [reply proj/src/sub2/bin/foo_main.ml]: My old self is not convinced. I've heard this style is mandatory in Rust... but I don't see what purpose it serves.

Comment (samoht) [reply proj/src/sub2/bin/foo_main.ml]: Which style? Turning on warning-as-error in development mode?

Comment (mjambon) [reply proj/src/sub2/bin/foo_main.ml]: I see the value of `--dev` giving us warnings that are relevant to the developer but might be useless and confusing to the simple user.

What I don't like is having to prefix unused variables with an underscore, with the exception of let bindings.

**Comment by mjambon** [comment]
We just did the migration to dune, which now uses the dev settings by default.

**Pull request closed without merging**